docs/user/openstack/deploy_sriov_workers.md

```diff
@@ -0,0 +1,378 @@
+- A flavor with at least 16 GB memory, 4 vCPUs, and 25 GB storage space
+
+Additionally, for clusters that use single-root input/output virtualization (SR-IOV), RHOSP compute nodes require a flavor that supports [huge pages](huge-pages).
+Deploying worker nodes with SR-IOV networks is supported as a post install operation in the installer for both IPI and UPI workflows. Once you have verified that your OpenStack cluster can support SR-IOV in OpenShift, and have succesfully installed an OpenShift cluster that meets the [minimum requirements](README.md#openstack-requirements), you can follow these steps and example workflows to create worker nodes with SR-IOV NICs.
+
+## Preparing your OpenShift Cluster to use SR-IOV Networks
+
+Before you use single root I/O virtualization (SR-IOV) on a cluster that runs on Red Hat OpenStack Platform (RHOSP), make the RHOSP metadata service mountable as a drive and enable the No-IOMMU Operator for the virtual function I/O (VFIO) driver.
+
+### Enabling the RHOSP metadata service as a mountable drive
+
+Next, you will need to create a machine config that enables the SR-IOV operator to get the UUID of networks from OpenStack. This machine config will allow the OpenStack metadata service to be available as a mountable drive.
+
+```yaml
+kind: MachineConfig
+apiVersion: machineconfiguration.openshift.io/v1
+metadata:
+  name: 20-mount-config
+  labels:
+    machineconfiguration.openshift.io/role: worker
+spec:
+  osImageURL: ''
+  config:
+    ignition:
+      version: 2.2.0
+    systemd:
+      units:
+        - name: create-mountpoint-var-config.service
+          enabled: true
+          contents: |
+            [Unit]
+            Description=Create mountpoint /var/config
+            Before=kubelet.service
+
+            [Service]
+            ExecStart=/bin/mkdir -p /var/config
+
+            [Install]
+            WantedBy=var-config.mount
+
+        - name: var-config.mount
+          enabled: true
+          contents: |
+            [Unit]
+            Before=local-fs.target
+            [Mount]
+            Where=/var/config
+            What=/dev/disk/by-label/config-2
+            [Install]
+            WantedBy=local-fs.target
+```
+
+### Enabling the No-IOMMU feature for the RHOSP VFIO driver
+
+You can apply a machine config to your machine pool that enables the No-IOMMU feature for the Red Hat OpenStack Platform (RHOSP) virtual function I/O (VFIO) driver. The RHOSP vfio-pci driver requires this feature.
+
+```yaml
+kind: MachineConfig
+apiVersion: machineconfiguration.openshift.io/v1
+metadata:
+  name: 99-vfio-noiommu
+  labels:
+    machineconfiguration.openshift.io/role: worker
+spec:
+  osImageURL: ''
+  config:
+    ignition:
+      version: 2.2.0
+    storage:
+      files:
+      - filesystem: root
+        path: "/etc/modprobe.d/vfio-noiommu.conf"
+        contents:
+          source: data:text/plain;charset=utf-8;base64,b3B0aW9ucyB2ZmlvIGVuYWJsZV91bnNhZmVfbm9pb21tdV9tb2RlPTEK
```

Comment:
This data says: 
"options vfio enable_unsafe_noiommu_mode=1" 
maybe we should put that somewhere

Reply:
I am not sure what you are referencing?

Reply:
The MachineConfig creates the vfio-noiommu.conf file with the contents:

options vfio enable_unsafe_noiommu_mode=1

Maybe steal an explanation from the dpdk docs...

Linux kernel >= 4.5 and DPDK 16.04 support a new VFIO mode which allows VFIO to be used without IOMMU. While this is just as unsafe as using UIO, it does make it possible to use VFIO in situations where IOMMU is not available and uio_pci_generic does not work either, such as virtual hosts.

The no-IOMMU mode is off by default and must be specifically enabled.

[dpdk](https://dpdk-guide.gitlab.io/dpdk-guide/setup/binding.html
)

The noiommu option will need to be used until the G35 version of QEMU is included with OpenStack.  However, this is a detail probably not needed by the customer?